docs/rules/rules.md

```diff
@@ -0,0 +1,316 @@
+*   To add a development dependency, use the `pub` tool, if it is available,
+    with `dev:<package name>`. otherwise run
+    `flutter pub add dev:<package_name>`.
+*   To add a dependency override, use the `pub` tool, if it is available,
+    with `override:<package name>:1.0.0`. otherwise run
+    `flutter pub add override:<package_name>:1.0.0`.
+
+## Code Quality
+*   Adhere to maintainable code structure and separation of concerns (e.g., UI
+    logic separate from business logic).
+*   Adhere to meaningful and consistent naming conventions.
+
+## Dart Best Practices
+*   Follow the official Effective Dart guidelines
+    (https://dart.dev/effective-dart)
+*   Define related classes within the same library file. For large libraries,
+    export smaller, private libraries from a single top-level library.
+*   Group related libraries in the same folder.
+*   Add documentation comments to all public APIs, including classes,
+    constructors, methods, and top-level functions.
+*   Write clear comments for complex or non-obvious code. Avoid over-commenting.
+*   Don't add trailing comments.
+*   Ensure proper use of `async`/`await` for asynchronous operations with robust
+    error handling.
+*   Use pattern matching features where they simplify the code.
+
+## Flutter Best Practices
+*   Widgets (especially `StatelessWidget`) are immutable; when the UI needs to
+    change, Flutter rebuilds the widget tree.
+*   Prefer composing smaller widgets over extending existing ones.
+*   Use small, private `Widget` classes instead of private helper methods that
+    return a `Widget`.
+*   Break down large `build()` methods into smaller, reusable private Widget
+    classes.
+*   Use `ListView.builder` to create lazy-loaded lists for performance.
+*   Use `const` constructors for widgets and in `build()` methods whenever
+    possible to optimize performance.
+*   Avoid performing expensive operations, like network calls or complex
+    computations, directly within `build()` methods.
+
+## Application Architecture
+*   Aim for separation of concerns similar to MVC/MVVM, with defined Model,
+    View, and ViewModel/Controller roles.
+
+### State Management
+*   Default to Flutter's built-in state management solutions. Do not use a
+    third-party package unless explicitly requested.
+*   Use `Streams` and `StreamBuilder` for handling a sequence of asynchronous
+    events.
+*   Use `Futures` and `FutureBuilder` for handling a single asynchronous
+    operation that will complete in the future.
+*   Use `ValueNotifier` with `ValueListenableBuilder` for simple, local state
+    that involves a single value.
+
+```dart
+// Define a ValueNotifier to hold the state.
+final ValueNotifier<int> _counter = ValueNotifier<int>(0);
+
+// Use ValueListenableBuilder to listen and rebuild.
+ValueListenableBuilder<int>(
+  valueListenable: _counter,
+  builder: (context, value, child) {
+    return Text('Count: $value');
+  },
+);
+```
```

Comment:
nit: For proper Markdown formatting, this entire block should be indented four spaces so it's part of the bullet above.